app/lib/common/providers/notifiers/client_pref_notifier.dart

```diff
@@ -7,40 +7,36 @@
 ///
 library;
 
+import 'dart:async';
+
 import 'package:acter/common/providers/common_providers.dart';
-import 'package:acter/features/home/providers/client_providers.dart';
 import 'package:acter_flutter_sdk/acter_flutter_sdk.dart';
 import 'package:riverpod/riverpod.dart';
 import 'package:shared_preferences/shared_preferences.dart';
 
 /// The type parameter `T` is the type of value that will
 /// be persisted in [SharedPreferences].
 ///
-/// To update the value, use the [update()] function.
+/// To update the value, use the [set()] function.
 /// Direct assignment to state cannot be used.
 ///
 /// ```dart
-/// await watch(booPrefProvider.notifier).update(v);
+/// await watch(booPrefProvider.notifier).set(v);
 /// ```
 ///
 ///
-class PrefNotifier<T> extends StateNotifier<T> {
-  PrefNotifier(this.prefKey, this.defaultValue) : super(defaultValue) {
-    _init();
-  }
+class PrefNotifier<T> extends Notifier<T> {
+  PrefNotifier(this.prefKeySuffix, this.defaultValue);
 
+  final String prefKeySuffix;
+  late String deviceId;
   late SharedPreferences prefs;
-  String prefKey;
-  T defaultValue;
 
-  void _init() async {
-    prefs = await sharedPrefs();
-    final newValue = (prefs.get(prefKey) as T? ?? defaultValue);
-    state = newValue;
-  }
+  String get prefKey => '$deviceId-$prefKeySuffix';
+  T defaultValue;
 
   /// Updates the value asynchronously.
-  Future<void> update(T value) async {
+  Future<void> set(T value) async {
```

Comment:
rename due to `update` already being defined by `Notifier` itself with a different signature.